src/axom/primal/geometry/Polygon.hpp

```diff
@@ -79,6 +80,30 @@ class Polygon
   /// Retrieves the vertex at index idx
   const PointType& operator[](int idx) const { return m_vertices[idx]; }
 
+  /*! 
+   * \brief Robustly returns the normal of the polygon (not normalized)
+   * \pre This function is only valid when NDIMS = 3 and the polygon is valid
+   * \return normal polygon normal vector
+   */
+  template <int TDIM = NDIMS>
+  typename std::enable_if<TDIM == 3, VectorType>::type normal() const
+  {
+    SLIC_ASSERT(isValid());
+    const int nverts = numVertices();
+
+    VectorType normal, v0(m_vertices[0], m_vertices[1]), v1;
+
+    // Iterate over each pair of vertices
+    for(int i = 2; i < nverts; ++i)
+    {
+      v1 = VectorType(m_vertices[0], m_vertices[i]);
+      normal += VectorType::cross_product(v0, v1);
+      v0 = v1;
+    }
```

Comment:
Actually, I naively copied this algorithm from the Graphics Gem about polyhedron in-out, but this isn't actually the best way to compute the normal vector. As I mentioned in the other comment, this algorithm is equivalent to computing normal vectors in each triangulation, but if you expand out the algebra of those cross products and cancel them out properly, you get an algorithm that iterates over consecutive pairs of edges instead of vectors from $V_0$ to $V_i$ and $V_{i+1}$. I'll update the algorithm in a bit. In the end, it'll make the algorithm independent of that choice of triangulation.

Reply:
Updated algorithm is in!